src/Endatix.Infrastructure/Integrations/Slack/SlackClient.cs

```diff
@@ -0,0 +1,112 @@
+    }
+
+    public async Task Handle(SubmissionCompletedEvent notification, CancellationToken cancellationToken)
+    {
+        Guard.Against.Null(_slackSettings.Active, nameof(_slackSettings.Active), "Slack settings must specify whether the integration is active.");
+        Guard.Against.NullOrEmpty(_slackSettings.EndatixHubBaseUrl, nameof(_slackSettings.EndatixHubBaseUrl), "EndatixHubBaseUrl must have a value within the Slack settings collection.");
+        Guard.Against.NullOrEmpty(_slackSettings.Token, nameof(_slackSettings.Token), "Token must have a value within the Slack settings collection.");
+        Guard.Against.NullOrEmpty(_slackSettings.ChannelId, nameof(_slackSettings.ChannelId), "ChannelId must have a value within the Slack settings collection.");
+
+        if (notification.Submission.IsComplete && (bool)_slackSettings.Active)
+        {
+
+            var submissionUrl = string.Format(SLACK_SUBMISSIONS_URL_TEMPLATE, _slackSettings.EndatixHubBaseUrl.TrimEnd('\\', '/'), notification.Submission.FormId);
+            Form? form;
+
+            using (var scope = _serviceProvider.CreateScope())
+            {
+                var repository = scope.ServiceProvider.GetRequiredService<IRepository<Form>>();
+                form = await repository.GetByIdAsync(notification.Submission.FormId, cancellationToken);
+            }
+
+            Guard.Against.Null(form, $"Unable to load submission {notification.Submission.Id}'s Form object.");
+
+            var message = string.Format(SLACK_MESSAGE_TEMPLATE, submissionUrl, form.Name);
+
+            await PostMessageAsync(_slackSettings.Token, _slackSettings.ChannelId, message);
+        }
+        else
+        {
+            if (!notification.Submission.IsComplete)
+            {
+                _logger.LogWarning($"SubmissionCompletedEvent raised on an incomplete submission with id {notification.Submission.Id}");
+            }
+
+        }
+    }
+
+    public async Task PostMessageAsync(string token, string channelId, string message)
+    {
+        Guard.Against.NullOrEmpty(channelId, nameof(channelId), "EndatixHubBaseUrl must have a value when posting to Slack.");
+        Guard.Against.NullOrEmpty(token, nameof(token), "Token must have a value  when posting to Slack.");
+        Guard.Against.NullOrEmpty(message, nameof(message), "Message must have a value when posting to Slack.");
+
+        var httpClient = _httpClientFactory.CreateClient();
+
+        httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {token}");
+
+        var payload = new
+        {
+            channel = channelId,
+            text = message
+        };
+
+        var content = new StringContent(
+            System.Text.Json.JsonSerializer.Serialize(payload),
+            Encoding.UTF8,
+            "application/json"
+        );
+
+        try
+        {
+            var response = await httpClient.PostAsync("https://slack.com/api/chat.postMessage", content);
```

Comment:
Looks like a good candidate that the URL is also set as a constant, like the others.

Reply:
Done